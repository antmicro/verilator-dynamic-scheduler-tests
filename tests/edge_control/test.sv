module t(/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;

   logic clk_copy;

   assign clk_copy = clk;

   initial begin
      $write("[%0t] start; clk == %b\n", $time, clk);
      @clk;
      $write("[%0t] any edge; clk == %b\n", $time, clk);
      @clk;
      $write("[%0t] any edge; clk == %b\n", $time, clk);
      @(negedge clk);
      $write("[%0t] negedge; clk == %b\n", $time, clk);
      @(negedge clk);
      $write("[%0t] negedge; clk == %b\n", $time, clk);
      @(posedge clk);
      $write("[%0t] posedge; clk == %b\n", $time, clk);
      @(posedge clk);
      $write("[%0t] posedge; clk == %b\n", $time, clk);
      @(posedge clk, negedge clk_copy);
      $write("[%0t] posedge or negedge; clk == %b\n", $time, clk);
      @(posedge clk or negedge clk_copy);
      $write("[%0t] posedge or negedge; clk == %b\n", $time, clk);
      $write("*-* All Finished *-*\n");
      $finish;
   end

endmodule
