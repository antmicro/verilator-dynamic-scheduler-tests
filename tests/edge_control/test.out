[0] start; clk == 0
[1] any edge; clk == 1
[2] any edge; clk == 0
[4] negedge; clk == 0
[6] negedge; clk == 0
[7] posedge; clk == 1
[9] posedge; clk == 1
[10] posedge or negedge; clk == 0
[11] posedge or negedge; clk == 1
*-* All Finished *-*
